src/linux/system.rs

```diff
@@ -520,14 +506,27 @@ impl SystemExt for System {
     }
 
     fn get_kernel_version(&self) -> Option<String> {
-        match read_to_string("/proc/version") {
-            Ok(s) => s.trim().split(' ').nth(2).map(String::from),
-            Err(_) => None,
+        let mut raw = std::mem::MaybeUninit::<libc::utsname>::zeroed();
+
+        if unsafe { libc::uname(raw.as_mut_ptr()) } == 0 {
+            let buffer: Vec<u8> = {
+                let info = unsafe { raw.assume_init() };
+
+                info.release
```

Comment:
It can be done like this too:

```rust
x.iter().filter(|c| c != 0).map(|c| *c as u8 as char).collect::<String>()
```

It allows to not have to allocate another `vec`.

Reply:
I considered this but it seemed less clean. 

FWIW, `String::from_utf8` reuses the `Vec<u8>` you pass in and doesn't re-allocate. Otherwise I would of collected directly into a `String`.

Reply:
I proposed another alternative just below. ;)

Reply:
I ended up choosing
```rust
info.release
  .iter()
  .map(|c| *c as u8)
  .filter(|c| *c != 0)
  .map(|c| c as char)
  .collect::<String>();
```

The other method would of required a second allocation as all of the bytes in `info.release` have to be converted from `c_char` to `u8` before they can be compared. Some Android platforms define `c_char` differently then others.

Reply:
Please remove the unneeded `.map(|c| *c as u8)` (or is there a specific reason for it?). If the issue is the conversion between `i8` and `char`, you can do it like this `c as u8 as char`.

Reply:
Its been removed. I had it that way to avoid `char::from(0)` in the filter, which is purely stylistic.

It is now this:
```rust
 let release = info
    .release
    .iter()
    .map(|c| *c as u8 as char)
    .filter(|c| *c != char::from(0))
    .collect::<String>();
```

Reply:
Still please put the filter before the map, I'm not sure it's super great to have `0 as char`...

Reply:
Sure, no problem. The filter has been moved.